### PR TITLE
Introduce Raw Query Editor Admin Console UI

### PR DESCRIPTION
**Pull request by 121watts**
Connect #823 
Connect #611 
Connect #914 

### The Problem
Poor feedback when typing InfluxQL in the query builder.  There are 4 states we need to handle / display to the user:

1) Success!  200 from server with results chock full o data.
2) A syntactically correct InfluxQL statement that returns no data.  This can happen because the data isn't there or the measurement, field, tag being queried doesn't exist.  Both of these situations return a 200 with `{results:[]}`.
3) A query that returns a 200 and an error in the results array: `{results:[{error: '...'}]}`.  This occurs with syntactically valid InfluxQL, but the user forgot to specify something like a database in their query.
4) A query that returns a 400 and an error from the Chronograf server.  This occurs with a syntactically invalid InfluxQL query.  I'm treating this as a failure.

### The Solution
Give the users some feedback.  In the code I treat 1 as a success, 2 & 3 as a warning, and 4 as a failure.

### Notes

* There was no use for the `MultiTable` component.  The tabbed view in tables just caused confusion.  So, I took it out.

* A lot more refactoring could get done in the DE and its children.  I would like to do some more after this has been reviewed.  

* It worries me how coupled the DE and Overlay Technologies are now intertwined.

### CODE REVIEW

**Comment by alexpaxton** [comment]
+1 to intertwined DE / Overlay tech, starting to give me design headache

File: ui/src/data_explorer/components/Visualization.js [hunk 1]
```diff
@@ -40,13 +47,71 @@ const Visualization = React.createClass({
...
+    )
+  },
+
+  renderVisualization(view, queries, heightPixels, onEditRawStatus, activeQueryIndex) {
+    const activeQuery = queries[activeQueryIndex]
+    const defaultQuery = queries[0]
+    const q = activeQuery || defaultQuery
+
+    switch (view) {
+      case GRAPH:
```
Comment (lukevmorris) [inline ui/src/data_explorer/components/Visualization.js]: Do we care about condensing this into:
```
switch (view) {
  case TABLE:
    ...
  case GRAPH:
  default:
    this.renderGraph(queries)
```

File: ui/src/data_explorer/components/Visualization.js [hunk 2]
```diff
@@ -40,13 +47,71 @@ const Visualization = React.createClass({
...
+        return this.renderGraph(queries)
+      case TABLE:
+        return (q ? <Table query={activeQuery || defaultQuery} height={heightPixels} onEditRawStatus={onEditRawStatus} /> : <div className="generic-empty-state">Enter your query below</div>)
```
Comment (cryptoquick) [inline ui/src/data_explorer/components/Visualization.js]: This should be written better as a conditional. Then you wouldn't need the q const. You'd just perform the conditional logic in q as the conditional. If you're having trouble putting a conditional block in your case, make it sure there are blocks for the cases.

File: ui/spec/data_explorer/reducers/queryConfigSpec.js [hunk 3]
```diff
@@ -10,7 +10,8 @@ import {
   groupByTime,
   toggleTagAcceptance,
   updateRawQuery,
-} from 'src/data_explorer/actions/view'
+  editRawQueryStatus,
+} from 'src/data_explorer/actions/view';
```
Comment (lukevmorris) [inline ui/spec/data_explorer/reducers/queryConfigSpec.js]: 🔫

File: ui/src/data_explorer/components/Table.js [hunk 4]
```diff
@@ -31,50 +35,74 @@ const ChronoTable = React.createClass({
...
+    if (this.props.query.text === nextProps.query.text) {
+      return
+    }
+
+    this.fetchCellData(nextProps.query)
+  },
+
+
+  async fetchCellData(query) {
+    if (!query || !query.text) {
```
Comment (lukevmorris) [inline ui/src/data_explorer/components/Table.js]: I thought JS `||` short-circuits if the left side is true? So
`if !query` or
`if query || !query.text`

File: ui/src/data_explorer/components/Visualization.js (hunk 2 above)
Comment (lukevmorris) [inline ui/src/data_explorer/components/Visualization.js]: agree -- this line's a bit long to understand easily

File: ui/src/data_explorer/components/QueryEditor.js [hunk 5]
```diff
@@ -89,9 +91,9 @@ const QueryEditor = React.createClass({
 
   renderQuery() {
     const {query, timeRange} = this.props
-    const statement = query.rawText || buildInfluxQLQuery(timeRange, query) || `SELECT "fields" FROM "db"."rp"."measurement"`
+    const statement = query.rawText || buildInfluxQLQuery(timeRange, query) || `Select a database, measurement, and field below.`
 
-    if (!query.rawText) {
+    if (typeof query.rawText !== 'string') {
```
Comment (121watts) [inline ui/src/data_explorer/components/QueryEditor.js]: Exactly, it makes rawText a trinary.

File: ui/src/data_explorer/components/RawQueryEditor.js [hunk 6]
```diff
@@ -25,10 +26,10 @@ const RawQueryEditor = React.createClass({
 
   handleKeyDown(e) {
     if (e.keyCode === ENTER) {
+      e.preventDefault()
       this.handleUpdate()
-      this.editor.blur()
     } else if (e.keyCode === ESCAPE) {
-      this.setState({value: this.props.query.rawText}, () => {
+      this.setState({value: this.state.value}, () => {
```
Comment (121watts) [inline ui/src/data_explorer/components/RawQueryEditor.js]: It does indeed :)

File: ui/src/data_explorer/components/Table.js [hunk 7]
```diff
@@ -31,50 +35,74 @@ const ChronoTable = React.createClass({
...
     // second param is db, we want to leave this blank
-    fetchTimeSeries(query.host, undefined, query.text).then((resp) => {
-      const cellData = _.get(resp.data, ['results', '0', 'series', '0'], false)
-      if (!cellData) {
-        return this.setState({isLoading: false})
+    try {
+      const {data} = await fetchTimeSeries(query.host, undefined, query.text)
+      this.setState({isLoading: false})
+
+      const results = _.get(data, ['results', '0'], false)
```
Comment (121watts) [inline ui/src/data_explorer/components/Table.js]: You got it!

File: ui/src/data_explorer/components/Table.js [hunk 8]
```diff
@@ -31,50 +35,74 @@ const ChronoTable = React.createClass({
...
-  },
+      if (_.isEmpty(results)) {
+        this.setState({cellData: emptyCells})
+        return onEditRawStatus(query.id, {warn: 'Your query is syntactically correct but returned no results'})
+      }
 
-  componentDidMount() {
-    this.fetchCellData(this.props.query)
-  },
+      const warn = _.get(results, 'error', false)
```
Comment (121watts) [inline ui/src/data_explorer/components/Table.js]: Yea, this is confusing and an unfortunate part of Influx's error handling and response bodies.

File: ui/src/data_explorer/components/Visualization.js (hunk 1 above)
Comment (121watts) [inline ui/src/data_explorer/components/Visualization.js]: love it

**Pull request merged**